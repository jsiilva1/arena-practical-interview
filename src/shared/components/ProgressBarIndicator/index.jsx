/**
 * External Dependencies
 */
import React from 'react';

/**
 * Internal Dependencies
 */
import TopBarProgressIndicator from 'react-topbar-progress-indicator';

/**
 * BarProgressIndicator Component
 * 
 * Render a top progress bar on load page
 *
 * @param {Object} props component props.
 * 
 * @return {Object} component.
 */
const ProgressBarIndicator = (props) => {
  const { loading } = props;

  TopBarProgressIndicator.config({
    barColors: {
      0: '#9B53E8',
    },
    shadowBlur: 0,
    barThickness: 2,
  });

  return (
    <>
      {!loading && (
        <TopBarProgressIndicator />
      )}
    </>
  );
};

export default ProgressBarIndicator;
