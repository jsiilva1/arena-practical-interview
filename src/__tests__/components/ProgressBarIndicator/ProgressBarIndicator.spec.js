import React from 'react';
import { mount } from 'enzyme';

import ProgressBarIndicator from '../../../shared/components/ProgressBarIndicator';

describe('ProgressBarIndicator Component', () => {
  it('should render the ProgressBarIndicator Component correctly if the page is loading', () => {
    const wrapper = mount(<ProgressBarIndicator loading={true} />);
    
    expect(wrapper).toMatchSnapshot();
  });
});
