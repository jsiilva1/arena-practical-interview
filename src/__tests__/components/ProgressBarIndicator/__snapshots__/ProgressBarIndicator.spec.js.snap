// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBarIndicator Component should render the ProgressBarIndicator Component correctly if the page is loading 1`] = `
<ProgressBarIndicator
  loading={true}
/>
`;
